src/Microsoft.AspNetCore.ResponseCompression/MimeTypeFilter.cs

```diff
@@ -0,0 +1,207 @@
+        /// </summary>
+        /// <param name="mimeType">MIME type to check.</param>
+        /// <returns>True if the response with passed MIME type should be compressed, otherwise false.</returns>
+        public bool ShouldCompress(string mimeType)
+        {
+            bool? shouldCompress;
+
+            //check exact match (type/subtype)
+            shouldCompress = ShouldCompressExact(mimeType);
+
+            if (shouldCompress.HasValue)
+            {
+                return shouldCompress.Value;
+            }
+
+            //check partial match (type/*)
+            int slashPos = mimeType.IndexOf('/');
+
+            if (slashPos > -1)
+            {
+                string wildcardMimeType = $"{mimeType.Substring(0, slashPos + 1)}*";
+                shouldCompress = ShouldCompressExact(wildcardMimeType);
+
+                if (shouldCompress.HasValue)
+                {
+                    return shouldCompress.Value;
+                }
+            }
+
+            //check wildcard (*/*)
+            shouldCompress = ShouldCompressExact("*/*");
+
+            if (shouldCompress.HasValue)
+            {
+                return shouldCompress.Value;
+            }
+
+            //no matches and no "*/*" entry in either list - do not compress
+            return false;
+        }
+
+        private MimeTypeFilter Add(ISet<string> set, IEnumerable<string> mimeTypes)
+        {
+            if (mimeTypes != null)
+            {
+                foreach (string mimeType in mimeTypes)
+                {
+                    if (!IsValid(mimeType))
+                    {
+                        throw new ArgumentException(
+                            $@"MIME type ""{mimeType}"" does not match the format ""<type>/<sybtype>"", ""<type>/*"" or ""*/*"".",
+                            nameof(mimeTypes)
+                        );
+                    }
+
+                    set.Add(mimeType);
+                }
+            }
+
+            HasBeenPopulated = true;
+            return this;
+        }
+
+        private MimeTypeFilter Remove(ISet<string> set, IEnumerable<string> mimeTypes)
+        {
+            if (mimeTypes != null)
+            {
+                set.ExceptWith(mimeTypes);
+            }
+
+            HasBeenPopulated = true;
+            return this;
+        }
+
+        private MimeTypeFilter Clear(ISet<string> set)
+        {
+            set.Clear();
+            HasBeenPopulated = true;
+            return this;
+        }
+
+        //This is NOT a proper MIME type validation. It only checks conformance to the format
+        //"type/sybtype", "type/*" or "*/*", for the purpose of being used in the lists of MIME
+        //types that should and should not be compressed.
+        private static bool IsValid(string mimeType) =>
+            Regex.IsMatch(mimeType, @"^([^\s\*/]+/([^\s\*/]+|\*)|\*/\*)$", RegexOptions.Compiled);
```

Comment:
Don't use regex for validation. We could use MediaTypeHeaderValue.TryParse, or just not worry about the format, we're only matching against it, not sending it to the client.

Reply:
`MediaTypeHeaderValue.TryParse` is less restrictive than what I'm trying to achieve. E.g. it accepts `*/subtype` or `type/sub*`. If I remove validation, entries like that will be just ignored, which can be an element of surprise. If you still think it's better to not validate, will be happy to remove.